src/mongocxx/include/mongocxx/v1/events/server_heartbeat_failed.hpp

```diff
@@ -32,7 +39,39 @@ namespace events {
 ///
 /// @attention This feature is experimental! It is not ready for use!
 ///
-class server_heartbeat_failed {};
+class server_heartbeat_failed {
+   private:
+    void const* _impl; // mongoc_apm_server_heartbeat_failed_t const*
+
+   public:
+    ///
+    /// Return the error message.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL(std::string) message() const;
+
+    ///
+    /// Return the execution time of the event.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL(std::int64_t) duration() const;
+
+    ///
+    /// Return the hostname for the connection used by the command.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL(bsoncxx::v1::stdx::string_view) host() const;
+
+    ///
+    /// Return the port number for the connection used by the command.
+    ///
+    MONGOCXX_ABI_EXPORT_CDECL(std::uint16_t) port() const;
+
+    ///
+    /// Return true when this heartbeat event is for an awaitable "hello" (instead of a "legacy" hello).
```

Comment:
```suggestion
    /// Return true when this heartbeat event is for an awaitable "hello" (instead of a "polling" hello).
```

I expect legacy hello refers to the `isMaster` command. Quoting [Server Monitoring](https://github.com/mongodb/specifications/blob/d41d48b90ef22aedae934dd22f3a21c65b5a13bc/source/server-discovery-and-monitoring/server-monitoring.md#monitoring) spec:

> The client monitors servers using the hello or legacy hello commands. In MongoDB 4.4+, a monitor uses the [Streaming Protocol](#streaming-protocol) to continuously stream hello or legacy hello responses from the server. In MongoDB \<= 4.2, a monitor uses the [Polling Protocol](#polling-protocol)